src/SchedulerSubsystem/Scheduler.java

```diff
@@ -62,6 +79,49 @@ else if (event instanceof DestinationEvent)
             throw new InvalidTypeException("Event type received cannot be handled by this subsystem.");
     }
 
+    /**
+     * Filters destination floors from destination events.
+     *
+     * @param events Set of destination events.
+     * @return Set of destination floors.
+     */
+    private Set<Integer> filterDestinationFloors(Set<DestinationEvent> events) {
+        Set<Integer> destinationFloors = new HashSet<>();
+        for (DestinationEvent e: events) {
+            destinationFloors.add(e.destinationFloor());
+        }
+        return destinationFloors;
+    }
+
+    /**
+     * Takes union of two destination event sets.
+     *
+     * @param A Set of destination events.
+     * @param B Set of destination events.
+     * @return Union of both sets.
+     */
+    private Set<DestinationEvent> unionOfDestinationEvents(
+            Collection<DestinationEvent> A,
+            Collection<DestinationEvent> B) {
+        HashSet<DestinationEvent> union = new HashSet<DestinationEvent>();
+        for (DestinationEvent a: A) {
+            if (B.contains(a)) {
+                union.add(a);
+            }
+        }
+        return union;
+    }
+
+    /**
+     * Sets destination events of this scheduler.
+     *
+     * @param events Set of destination events to be set.
+     */
+    public void setFloorRequests(Set<DestinationEvent> events) {
```

Comment:
This seems purely like a testing function, as I've said before I think that's a bad idea. Especially setters.